llama.cpp

```diff
@@ -11220,7 +11409,8 @@ int32_t llama_tokenize(
                      int32_t   n_max_tokens,
                         bool   add_bos,
                         bool   special) {
-    auto res = llama_tokenize_internal(model->vocab, std::string(text, text_len), add_bos, special);
+    bool add_space_prefix = false ? model->arch == LLM_ARCH_INTERNLM2 : true;
```

Comment:
The condition is wrong:

```suggestion
    bool add_space_prefix = model->arch == LLM_ARCH_INTERNLM2 ? true : false;
```

Is this change necessary? Seems strange to add prefix just for a specific model architecture. Can this be fixed outside of `llama.cpp`?

Reply:
Thanks for your suggestion! In terms of deciding the value for `add_dummy_prefix`, I went through the SentencePiece code and found that the tokenizer's `add_dummy_prefix` value is stored [there](https://github.com/google/sentencepiece/blob/53de76561cfc149d3c01037f0595669ad32a5e7c/src/builtin_pb/sentencepiece_model.pb.h#L1315). And we can get the value from `tokenizer.model` through the python interface:
```python
from sentencepiece import sentencepiece_model_pb2 as model
m = model.ModelProto()
m.ParseFromString(open(TOKENIZER_PATH, "rb").read())
print(m.normalizer_spec.add_dummy_prefix)
```

So, maybe when we're doing the checkpoint conversion in convert-hf-to-gguf.py, we could consider adding a new key-value field using gguf_writer. For example, we might use something like 
```python
gguf_writer.add_add_dummy_prefix(add_dummy_prefix: bool).
```
If you're good with it, I'll make a change so that we're not tying the decision on `add_dummy_prefix` to any specific model type @ggerganov

Reply:
@SolenoidWGT Yes, I think this is a good solution. Please make the change

Reply:
> @SolenoidWGT Yes, I think this is a good solution. Please make the change
Done.